src/sentry/static/sentry/app/views/organizationIntegrations/index.jsx

```diff
@@ -148,7 +146,55 @@ class OrganizationIntegrations extends AsyncComponent {
           onReinstall={this.onInstall}
           enabledPlugins={this.enabledPlugins}
         />
-      ));
+      </IntegrationRow>
+    );
+  }
+
+  renderSentryApps(apps, key) {
+    const {appInstalls} = this.state;
+
+    return (
+      <IntegrationRow key={`row-${key}`}>
+        <SentryAppInstallations
+          key={key}
+          orgId={this.props.params.orgId}
+          installs={appInstalls}
+          applications={apps}
+        />
+      </IntegrationRow>
+    );
+  }
+
+  renderBody() {
+    const {reloading, applications, appInstalls} = this.state;
+
+    const installedProviders = this.providers
```

Comment:
There's kind of a lot of duplicated and/or very similar code, we could use `lodash.partition` to cut back some code. Downside of this is that readers will need to look up what `partition` does.  (Current code is fine too though).

```
const [installedProviders, uninstalledProviders] = _.partition(this.providers.map(p => [p, this.renderProvider(p)]), ([provider,]) => provider.isInstalled);
```